paddle/framework/scope.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+#include "paddle/framework/variable.h"
+#include "paddle/utils/Error.h"
+
+namespace paddle {
+namespace framework {
+
+const static Error AlreadyCreated("Variable has already been created");
+
+/**
+ * Scope is an association of a name to Variable. All variables belong to
+ * `Scope`. You need to specify a scope to run a Net, i.e., `net.Run(&scope)`.
+ * One net can run in different scopes and update different variable in the
+ * scope.
+ */
+class Scope {
+ public:
+  Scope() {}
+
+  explicit Scope(const std::shared_ptr<Scope>& scope) : parent_(scope) {}
+
+  ~Scope() {}
+
+  // Create Variable in this Scope. Return error if Variable already been
+  // created.
+  Error __must_check CreateVariable(const std::string& name);
```

Comment:
I checked the definition of `__must_check`. It means "must use the returned value", and has nothing about "check".

Also, according to Google C++ style, a macro should be all captital, e.g., `MUST_USED_RETVALUE`. It breaks the style.

Should we remove `paddle/platform/must_check.h`?